dlt/destinations/dataset/ibis_relation.py

```diff
@@ -0,0 +1,161 @@
+    from dlt.common.libs.ibis import Expr
+except MissingDependencyException:
+    Expr = Any
+
+# map dlt destination to sqlglot dialect
+DIALECT_MAP = {
+    "dlt.destinations.duckdb": "duckdb",  # works
+    "dlt.destinations.motherduck": "duckdb",  # works
+    "dlt.destinations.clickhouse": "clickhouse",  # works
+    "dlt.destinations.databricks": "databricks",  # works
+    "dlt.destinations.bigquery": "bigquery",  # works
+    "dlt.destinations.postgres": "postgres",  # works
+    "dlt.destinations.redshift": "redshift",  # works
+    "dlt.destinations.snowflake": "snowflake",  # works
+    "dlt.destinations.mssql": "tsql",  # works
+    "dlt.destinations.synapse": "tsql",  # works
+    "dlt.destinations.athena": "trino",  # works
+    "dlt.destinations.filesystem": "duckdb",  # works
+    "dlt.destinations.dremio": "presto",  # works
+    # NOTE: can we discover the current dialect in sqlalchemy?
+    "dlt.destinations.sqlalchemy": "mysql",  # may work
+}
+
+# NOTE: some dialects are not supported by ibis, but by sqlglot, these need to
+# be transpiled with a intermediary step
+TRANSPILE_VIA_MAP = {
+    "tsql": "postgres",
+    "databricks": "postgres",
+    "clickhouse": "postgres",
+    "redshift": "postgres",
+    "presto": "postgres",
+}
+
+
+# TODO: provide ibis expression typing for the readable relation
+class ReadableIbisRelation(BaseReadableDBAPIRelation):
+    def __init__(
+        self,
+        *,
+        readable_dataset: ReadableDBAPIDataset,
+        expression: Expr = None,
+    ) -> None:
+        """Create a lazy evaluated relation to for the dataset of a destination"""
+        super().__init__(readable_dataset=readable_dataset)
+        self._expression = expression
+
+    @property
+    def query(self) -> Any:
+        """build the query"""
+
+        from dlt.common.libs.ibis import ibis, sqlglot
+
+        destination_type = self._dataset._destination.destination_type
+        target_dialect = DIALECT_MAP[destination_type]
+
+        # render sql directly if possible
+        if target_dialect not in TRANSPILE_VIA_MAP:
+            return ibis.to_sql(self._expression, dialect=target_dialect)
+
+        # here we need to transpile first
+        transpile_via = TRANSPILE_VIA_MAP[target_dialect]
+        sql = ibis.to_sql(self._expression, dialect=transpile_via)
+        sql = sqlglot.transpile(sql, read=transpile_via, write=target_dialect)[0]
+        return sql
+
+    @property
+    def columns_schema(self) -> TTableSchemaColumns:
+        return self.compute_columns_schema()
+
+    @columns_schema.setter
+    def columns_schema(self, new_value: TTableSchemaColumns) -> None:
+        raise NotImplementedError("columns schema in ReadableDBAPIRelation can only be computed")
+
+    def compute_columns_schema(self) -> TTableSchemaColumns:
+        """provide schema columns for the cursor, may be filtered by selected columns"""
+        # TODO: provide column lineage tracing with sqlglot lineage
+        return None
```

Comment:
maybe we could try some simple implementation:
1. we preserve table schema until first expression is made
2. we preserve table schema for `head` `limit` and `select` method
3. maybe we also recognize a few exceptions that do not modify schema ie `filter` and free form filter expressions as here: https://ibis-project.org/tutorials/ibis-for-pandas-users#filtering-rows

Reply:
What do you mean with "free form filter"?

Reply:
order_by and limit also are included, I have done this now.